libvast/src/command.cpp

```diff
@@ -32,29 +32,32 @@ command::~command() {
 }
 
 int command::run(caf::actor_system& sys, option_map& options,
-                 argument_iterator begin, argument_iterator end) {
-  VAST_TRACE(VAST_ARG(options), VAST_ARG("args", begin, end));
-  // Split the arguments.
-  auto args = caf::message_builder{begin, end}.move_to_message();
-  auto [local_args, subcmd, subcmd_args] = separate_args(args);
-  begin += local_args.size();
+                     argument_iterator begin, argument_iterator end) {
+  VAST_TRACE(VAST_ARG(std::string(name_)), VAST_ARG("args", begin, end),
+             VAST_ARG(options));
   // Parse arguments for this command.
-  auto res = local_args.extract_opts(opts_);
-  if (res.opts.count("help") != 0) {
-    std::cout << res.helptext << std::endl;
-    if (!nested_.empty()) {
-      std::cout << "\nSubcommands:\n";
-      for (auto& kvp : nested_)
-        std::cout << "  " << kvp.first << "\n";
-    }
-    std::cout << std::endl;
+  auto [state, position] = opts_.parse(options, begin, end);
+  bool has_subcommand;
+  switch(state) {
+    default:
+      std::cerr << parse_error(state, position, begin, end) << std::endl;
+      return EXIT_FAILURE;
+    case option_declaration_set::parse_state::successful:
+      has_subcommand = false;
+      break;
+    case option_declaration_set::parse_state::not_an_option:
+      if (position == end)
+        has_subcommand = false;
+      has_subcommand = true;
```

Comment:
You unconditionally set this to `true`. Did you miss an `else` here? In that case, there's no need to branch anyways: `has_subcommand = position != end`.